dali/test/python/test_pipeline_segmentation.py

```diff
@@ -0,0 +1,194 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import itertools
+import os
+from math import ceil, sqrt
+import numpy as np
+
+import nvidia.dali.fn as fn
+import nvidia.dali.math as math
+import nvidia.dali.types as types
+from nvidia.dali.pipeline import Pipeline
+from test_utils import get_dali_extra_path
+
+test_data_root = get_dali_extra_path()
+
+def check_bbox_random_crop_adjust_polygons(file_root, annotations_file, batch_size = 3, 
+                                           num_iters = 4, num_threads = 4, device_id = 0, seed=1234):
+    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=device_id, seed=seed)
+    with pipe:
+        # Read data from COCO
+        # ratio=True means both bboxes and masks coordinates will be relative to the image dimensions (range [0.0, 1.0])
+        inputs, in_bboxes, labels, in_polygons, in_vertices = \
+            fn.coco_reader(
+                file_root=file_root, annotations_file=annotations_file, shard_id=0, num_shards=1, 
+                ratio=True, ltrb=True, polygon_masks=True
+            )
+
+        # Generate a random crop. out_bboxes are adjusted to the crop window
+        slice_anchor, slice_shape, out_bboxes, labels, bbox_indices = \
+            fn.random_bbox_crop(
+                in_bboxes, labels,
+                aspect_ratio=[0.5, 2.0], thresholds=[0, 0.1, 0.3, 0.5, 0.7, 0.9], 
+                scaling=[0.3, 1.0], bbox_layout='xyXY', output_bbox_indices=True
+            )
+        # Crop the image
+        images = fn.image_decoder_slice(
+            inputs, slice_anchor, slice_shape,
+            device='mixed', axis_names='WH'
+        )
+
+        sel_polygons, sel_vertices = fn.segmentation.select_masks(
+            bbox_indices, in_polygons, in_vertices
+        )
+
+        # Adjust masks coordinates to the coordinate space of the cropped image
+        MT = fn.transforms.crop(from_start=slice_anchor, from_end=(slice_anchor + slice_shape))
+        out_vertices = fn.coord_transform(sel_vertices, MT=MT)
+
+        # Converting to absolute coordinates (demo purposes)
+        image_shape = fn.peek_image_shape(inputs, type=types.FLOAT)
+        h = fn.slice(image_shape, 0, 1, axes=[0])
+        w = fn.slice(image_shape, 1, 1, axes=[0])
+
+        # Original bboxes
+        bbox_x = fn.slice(in_bboxes, 0, 1, axes=[1])
+        bbox_y = fn.slice(in_bboxes, 1, 1, axes=[1])
+        bbox_X = fn.slice(in_bboxes, 2, 1, axes=[1])
+        bbox_Y = fn.slice(in_bboxes, 3, 1, axes=[1])
+        in_bboxes_abs = fn.cat(bbox_x * w, bbox_y * h, bbox_X * w, bbox_Y * h, axis=1)
```

Comment:
```suggestion
        in_bboxes_abs = fn.coord_transform(in_bboxes, fn.transforms.scale(scale=[w, h, w, h]))
```

Reply:
Despite the matrix-vector multiplication it should be cheaper just because there are much fewer operators.

Reply:
as we agreed offline, it's more straight forward the way it is now